### PR TITLE
DOC add build dependency to contributing guides

### PR DESCRIPTION
**Pull request by adrinjalali**
### Summary
Going through my first build and reading the contributing guides, I realized we're missing to mention that a dev environment requires `setuptools-rust`, which is picked up by `pip install ...` commands, but if a user does `python setup.py ...` they need to have them installed themselves. This PR adds a note for the dependencies.

I also noticed this in `pyproject.toml`:

```
[tool.black]
line-length = 100
target-version = ['py37', 'py38', 'py39', 'py310']
```

Do we want `py311` there (in this or a separate PR)?

### CODE REVIEW

**Review by jakelishman — COMMENTED** [review]
I'm ok to include this information, but I think it might be better just to tell people to install the packages listed in `pyproject.toml:build_system.requires` rather than listing them out here - it's less liable to get out-of-sync that way.  It might be better to put it after the line about `python setup.py build_rust` rather than before, because that's the only `setup.py` command somebody should really be running manually; everything else should go through `pip` (or some other PEP-517 builder).

**Comment by adrinjalali** [comment]
Sure, I moved it and removed the explicitly listed packages @jakelishman

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -256,6 +256,10 @@ If you'd like to use an editable install with an optimized binary you can
 run `python setup.py build_rust --release --inplace` after you install in
 editable mode to recompile the rust extensions in release mode.
 
+Note that in order to run `python seutp.py ...` commands you need have build
```
Comment (jakelishman) [inline CONTRIBUTING.md:259]:
```suggestion
Note that in order to run `python setup.py ...` commands you need have build
```

**Comment by jakelishman** [comment]
Oh, and to answer the other question: yeah, we could probably stand to update the listed versions for `black` there.  I'm not sure how much of an effect it has, but it's definitely better to tell it the correct thing.  If you want to open another PR for that, I'll approve it.

**Pull request merged**